socketDataHandler.js

```diff
@@ -76,39 +136,24 @@ class SocketDataHandler {
         } else if(this.state == 2) { //PostLoginHandshake
           switch(packet.packetID){
             case 0:
-              var loginStart = new LoginStart(packet)
-              log(`User ${loginStart.username} is logging in...`)
-              var loginSuccess = new LoginSuccess(loginStart.username)
-              this.socket.write(loginSuccess.loadIntoBuffer())
-              var joinGame = new JoinGame(loginStart.username)
-              this.socket.write(joinGame.loadIntoBuffer())
-              var spawnPosition = new SpawnPosition(0,3,0)
-              this.socket.write(spawnPosition.loadIntoBuffer())
-              for(var x = -4; x <= 4; x++){
-                for(var z = -4; z <= 4; z++){
-                  var loadChunk = new ChunkData(x,z)
-                  this.socket.write(loadChunk.loadIntoBuffer())
-                }
-              }
-              var playerPosition = new PlayerPosition(0,30,0,0,0)
-              this.socket.write(playerPosition.loadIntoBuffer())
-              this.state = 4
+              var username = this.processLogin(packet)
+              this.confirmLogin(username)
+              this.joinGame()
+              this.loadArea()
+              this.connectToPeers(username)
+              this.placePlayer()
               this.keepAliveInterval = setInterval(this.keepAlive.bind(this), keepAliveSendInterval)
+              this.state = 4
               break;
             default:
               log(`state 2: unexpected packet id ${packet.packetID}`)
               break;
           }
         } else if(this.state == 3) {
```

Comment:
State 3 now represents a login connection from another p2p server. We get to this state by indicating it in the handshake. We separate this from state 2 because in this case we only need to do a subset of the tasks that a server would do in the case of a direct login.